src/lib.rs

```diff
@@ -1,40 +1,77 @@
 use std::{
     ffi::OsString,
-    path::Path,
-    process::{Command, ExitStatus},
+    path::{Path, PathBuf},
+    process::{Command, Output},
 };
 
 /// Options provided for compiling binaries.
 struct CompileOption {
     /// File to execute when compile.
-    pub compiler: OsString,
+    pub compiler: PathBuf,
     /// Arguments passed to compiler process.
     pub args: Vec<OsString>,
 }
 
-/// Result of compilation.
```

Comment:
Should we alter `CompileResult` with `std::process::Output`? We could add metadata like time consumed for compilation (for Compile Time Limit Exceeded), structured compile error explanation (for example, what line causes a syntax error), etc.

Reply:
We should kill compile process to limit compile time (**Will change.**), therefore time consumed in compilation need not be measured directly. Compile error is guaranteed to be provided as just a byte sequence, so it structuring could be handled at web. I don't think we can get any other meaningful information from the running proccess than `Output`.

Reply:
ok then it's ok to use Output directly